src/irace/__init__.py

```diff
@@ -148,10 +198,37 @@ def set_initial(self, x):
         
     def run(self):
         """Returns a Pandas DataFrame, one column per parameter and the row index are the configuration ID."""
-        self.scenario['targetRunner'] = self.r_target_runner
-        res = self._pkg.irace(ListVector(self.scenario), self.parameters)
+        scenario_a = [None, None]
+        self.r_target_runner_parallel = make_target_runner_parallel(self.target_aq, self.target_rq, self._pkg.check_output_target_runner, scenario_a, self.target_runner, self.worker_count != 1)
+        self.scenario['targetRunnerParallel'] = self.r_target_runner_parallel
+
+        with localconverter(irace_converter_hack):
+            self.r_scenario = self._pkg.checkScenario(ListVector(self.scenario))
+        self.scenario = r_to_python(self.r_scenario)
+        self.scenario.pop('targetRunnerParallel', None)
+        scenario_a[0] = self.scenario
+        scenario_a[1] = self.r_scenario
```

Comment:
This may need a comment to explain what is going on: You pass `scenario_a` to `make_target_runner_parallel` using `None` values, but then assign values here? How does that work?

Reply:
`make_target_runner_parallel` needs to know the scenario but to make the scenario (pass the `checkScenario`), the `targetRunnerParallel` must be set, creating a chicken and egg problem. `scenario_a` kind of acts like a pointer. I can give it a null pointer first and then set the value later.